ipatests/test_integration/test_ipahealthcheck.py

```diff
@@ -0,0 +1,446 @@
+                "ipahealthcheck.ipa.host",
+                "--check",
+                "IPAHostKeytab",
+                "--output-file",
+                HEALTHCHECK_OUTPUT_FILE,
+            ],
+            raiseonerr=False,
+        )
+        contents = self.master.get_file_contents(
+            HEALTHCHECK_OUTPUT_FILE, encoding="utf-8"
+        )
+        data = json.loads(contents)
+        if result.returncode == 3:
+            assert data[0]["result"] == "ERROR"
+            assert data[0]["kw"]["msg"] == msg
+        else:
+            assert data[0]["result"] == "SUCCESS"
+        self.master.run_command(["systemctl", "start", dirsrv_service])
+
+    def test_source_ipahealthcheck_topology_IPATopologyDomainCheck(self):
+        """
+        Testcase checks behaviour of check IPATopologyDomainCheck in
+        source ipahealthcheck.ipa.topology on IPA Master
+        """
+        self.master.run_command(
+            [
+                "ipa-healthcheck",
+                "--source",
+                "ipahealthcheck.ipa.topology",
+                "--check",
+                "IPATopologyDomainCheck",
+                "--output-file",
+                HEALTHCHECK_OUTPUT_FILE,
+            ],
+            raiseonerr=False,
+        )
+        contents = self.master.get_file_contents(
+            HEALTHCHECK_OUTPUT_FILE, encoding="utf-8"
+        )
+        data = json.loads(contents)
+        for check in data:
+            assert check["result"] == "SUCCESS"
+            assert check["kw"]["suffix"] == "domain"
+            assert check["kw"]["suffix"] == "ca"
+
+    def test_source_ipa_roles_check_crlmanager(self):
+        """
+        This testcase checks the status of healthcheck tool
+        reflects correct information when crlgen is disabled
+        using ipa-crl-manage disable
+        """
+        msg = "The ipa-crlgen-enable command was successful"
+        cmd = self.master.run_command(["ipa-crl-manage", "disable"])
+        assert cmd.stdout_text == msg
+        self.master.run_command(
+            [
+                "ipa-healthcheck",
+                "--source",
+                "ipahealthcheck.ipa.roles",
+                "--check",
+                "IPACRLManagerCheck",
+                "--output-file",
+                HEALTHCHECK_OUTPUT_FILE,
+            ],
+            raiseonerr=False,
+        )
+        contents = self.master.get_file_contents(
+            HEALTHCHECK_OUTPUT_FILE, encoding="utf-8"
+        )
+        data = json.loads(contents)
+        for check in data:
+            assert check["result"] == "SUCCESS"
+            assert check["kw"]["key"] == "crl_manager"
+            assert check["kw"]["crlgen_enabled"] == "false"
+
+    def test_ipa_healthcheck_remove(self):
+        """
+        Testcase to check removal of healthcheck tool
+        """
+        cmd = self.master.run_command(
+            ["dnf", "remove", "-y", "freeipa-healthcheck"]
+        )
+        assert cmd.returncode == 0
```

Comment:
No need of this. `self.master.run_command(["dnf", "remove", "-y", "freeipa-healthcheck"])` will do.